### PR TITLE
feat: Implement the exported navigation component

### PR DESCRIPTION
**Pull request by mkarajohn**
Resolves [DSTRBTN-540]

<details open="true"><summary>Generated summary (powered by <a href="https://app.graphite.dev">Graphite</a>)</summary>

> ## TL;DR
> This pull request includes various changes to the codebase related to navigation and styling. It introduces new styled components, imports external libraries, and adds authentication-related functions and configurations. It also defines a React component called "Navigation" that renders a navigation menu based on the current URL and user role. Additionally, it includes changes to file structure and exports/imports of components and props.
> 
> ## What changed
> - Added styles for a navigation component, including wrapper, global navigation, and single icon container
> - Added styled components for creating a container for icons, including styles for container, individual icons, and their wrappers
> - Added CSS styles for various components such as buttons and links, including transitions, box shadows, and cursor styles
> - Added styled components for different types of links and buttons, as well as a styled component for a local navigation container
> - Imported various icons and components from external libraries
> - Added authentication-related functions and configurations
> - Imported components and functions related to navigation and authentication
> - Defined a dictionary for product icons
> - Created a React component called "Navigation" that renders a navigation menu based on props and current URL/user role
> - Defined a function that returns a JSX component based on device type and URL/user role
> - Rendered a list of icons for different apps, with tooltips and links
> - Defined a link that redirects the user based on a variable value
> - Exported and imported the Navigation component from a file called Navigation.ts, along with its associated props
> - Added new files: "Navigation/index.ts", "Navigation/types.ts", and "consts.ts"
> - Defined a location function, an array of menu options, and a constant for default navigation bar height
> - Exported the Navigation component and its associated props, as well as the TopBar component and its props
> 
> ## Why make this change
> This change was made to improve the navigation and styling of the application. The addition of new styled components and CSS styles allows for more consistent and visually appealing design across the application. The introduction of the "Navigation" component and related functionality provides a more dynamic and user-friendly navigation menu that adapts to the current URL and user role. The changes to file structure and exports/imports help organize the codebase and make it more modular and maintainable. Overall, these changes enhance the user experience and improve the overall quality of the application.
</details>

[DSTRBTN-540]: https://orfium.atlassian.net/browse/DSTRBTN-540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mkarajohn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #74** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/74" target="_blank">Graphite</a> 
    * **PR #75** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/75" target="_blank">Graphite</a> 
      * **PR #76** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/76" target="_blank">Graphite</a> 
        * **PR #77** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/77" target="_blank">Graphite</a> 
          * **PR #78** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/78" target="_blank">Graphite</a> 
            * **PR #79** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/79" target="_blank">Graphite</a> 
              * **PR #80** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/80" target="_blank">Graphite</a> 
                * **PR #81** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/81" target="_blank">Graphite</a> 
                  * **PR #82** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/82" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Orfium/toolbox/82?utm_source=stack-comment).

**Comment by mkarajohn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #74** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/74" target="_blank">Graphite</a> 
    * **PR #75** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/75" target="_blank">Graphite</a> 
      * **PR #76** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/76" target="_blank">Graphite</a> 
        * **PR #77** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/77" target="_blank">Graphite</a> 
          * **PR #78** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/78" target="_blank">Graphite</a> 
            * **PR #79** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/79" target="_blank">Graphite</a> 
              * **PR #80** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/80" target="_blank">Graphite</a> 
                * **PR #81** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/81" target="_blank">Graphite</a> 
                  * **PR #82** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/82" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Orfium/toolbox/82?utm_source=stack-comment).

File: src/ui/Navigation/Navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  useEffect(() => {
+    setExpanded(breakpoints.des1200);
+  }, [breakpoints.des1200]);
+
+  const isDesktop = breakpoints.des1200;
+  const adminNavigationIsActive = match?.url === adminNavigationURLSegment && userIsAdmin;
+
+  return (
+    <Wrapper
+      // onMouseEnter={() => !isDesktop && setExpanded(true)}
```
Comment (panvourtsis) [inline src/ui/Navigation/Navigation.tsx:76]: no need for comments

File: src/ui/Navigation/Navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                setExpanded((state) => !state);
+              }}
+              data-testid={'menu-handler'}
+            >
+              <Icon color={'primary'} name={'menu'} size={24} />
+            </BurgerButton>
+          )}
+        </SingleIconContainer>
+        <IconsContainer>
+          {orfiumProducts
```
Comment (panvourtsis) [inline src/ui/Navigation/Navigation.tsx:95]: we could initialize orfiumProducts with array to avoid this

**Review by panvourtsis — APPROVED** [review]
let me know about my optional comments, all others seems gtm

File: src/ui/Navigation/Navigation.tsx (hunk 1 above)
Comment (mkarajohn) [reply src/ui/Navigation/Navigation.tsx:76]: gone in one of the next prs

File: src/ui/Navigation/Navigation.tsx (hunk 2 above)
Comment (mkarajohn) [reply src/ui/Navigation/Navigation.tsx:95]: I am not completely opposed to this suggestion, but I like the distinction between "uninitialized" state from valid but empty state. 
Seeing `null` here implies it's uninitialized/still loading. 
Let's put this in the "revisit" list

**Comment by mkarajohn** [comment]
### Merge activity

* **Oct 24, 10:43 AM**: @mkarajohn started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Orfium/toolbox/82).

**Pull request merged**